src/GraphMap/Map.cs

```diff
@@ -0,0 +1,289 @@
+
+
+    List<Node> nodes;
+    List<Edge> edges;
+
+    public Node currentNode;
+
+    private Vector2 mapSize;
+
+    /// <summary>
+    /// Contains the symbol and color at a certain position.
+    /// </summary>
+    private Tuple<string, ConsoleColor>[,] mapSymbols;
+
+    public Map()
+    {
+        nodes = new List<Node>();
+        edges = new List<Edge>();
+        CreateMap();
+
+        // setup data to draw the map
+        mapSize = determineMapBoundaries();
+
+        mapSymbols = new Tuple<string, ConsoleColor>[(int)mapSize.X, (int)mapSize.Y];
+        fillMapSymbols();
+    }
+
+    public void CreateMap()
+    {
+        Node cityCentre = CreateNode(12, 12, ConsoleColor.Blue, "cityCentre", "The main square of a large city");
+        Node forest = CreateNode(16, 4, ConsoleColor.Green, "Forest", "A dark forest filled with trees");
+        Node mountains = CreateNode(2, 2, ConsoleColor.Gray, "Mountains", "Large mountain peaks with a cold climate surrounding them");
+        Node coast = CreateNode(9, 20, ConsoleColor.Yellow, "Coast", "Vast coastline seperating the land from the endless sea");
+
+        currentNode = cityCentre;
+
+        CreatEdge(cityCentre, forest, true, 10);
+        CreatEdge(cityCentre, mountains, false, 30);
+        CreatEdge(cityCentre, coast, false, 60);
+        CreatEdge(forest, mountains, true, 20);
+        CreatEdge(coast, forest, true, 50);
+        CreatEdge(mountains, coast, true, 30);
+    }
+
+    public Node CreateNode(int xPos, int yPos, ConsoleColor color, string name, string description)
+    {
+        Node node = new Node(xPos, yPos, color, name, description);
+        nodes.Add(node);
+        return node;
+    }
+
+    public void CreatEdge(Node startNode, Node destinationNode, bool biDirectional, int stepValue)
+    {
+        Edge edge = new Edge(startNode, destinationNode, biDirectional, stepValue);
+        edges.Add(edge);
+    }
+
+    public List<Edge> GetPaths()
+    {
+        List<Edge> paths = new List<Edge>();
+        foreach (Edge edge in edges)
+        {
+            if (edge.startNode == currentNode || (edge.destinationNode == currentNode && edge.biDirectional))
+            {
+                paths.Add(edge);
+            }
+
+        }
+        return paths;
+    }
+
+    /// <summary>
+    /// This function is responsible to caluclate the boundaries of the map according to the given nodes and their positions.
+    /// 
+    /// The map is always a little bit bigger than the actual nodes.
+    /// </summary>
+    /// <returns></returns>
+    private Vector2 determineMapBoundaries()
+    {
+        if (nodes == null) return new Vector2(0, 0);
+
+        int maxX = nodes.Max(node => node.xPos);
+        int maxY = nodes.Max(node => node.yPos);
+
+        return new Vector2(maxX + 5, maxY + 5);
+    }
+
+    /// <summary>
+    /// This function fills the double array with their according symbols.
+    /// </summary>
+    private void fillMapSymbols()

```

Comment:
```suggestion
    private void FillMapSymbols()

```

Reply:
Maybe also consider splitting the method into 3 smaller method each responsible for empty map creation, adding edges, adding locations.

Reply:
Done